src/main/java/seedu/address/logic/parser/FindExpressionParser.java

```diff
@@ -0,0 +1,288 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.logic.parser.FindFilterStringTokenizer.Token;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a parser that constructs and interprets expression trees
+ * from tokenized search queries. This parser helps in translating user
+ * input queries into predicates that can be applied to filter lists of
+ * persons.
+ * <p>
+ * Uses a basic recursive-descent parsing strategy, since for now the
+ * grammar is compact.
+ */
+public class FindExpressionParser {
```

Comment:
Would be nice if the javadocs can have some definition of the relationship between the expression, term and factors

e.g.:
```
expression :  term [OR term ...]
term: factor [AND factor ...]
factor: NOT factor | ( expression ) | CONDITION
```

Reply:
Good idea! I included a Backer-Naur form grammar in 9ca5392